packages/features/auth/lib/verifyEmail.ts

```diff
@@ -0,0 +1,66 @@
+import { randomBytes } from "crypto";
+import { getFeatureFlagMap } from "flags/server/utils";
+
+import { sendEmailVerificationLink } from "@calcom/emails/email-manager";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+import rateLimit from "@calcom/lib/rateLimit";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { prisma } from "@calcom/prisma";
+import { TRPCError } from "@calcom/trpc/server";
+
+const log = logger.getChildLogger({ prefix: [`[[Auth] `] });
+
+const limiter = rateLimit({
+  intervalInMs: 60 * 1000, // 1 minute
+});
+
+interface VerifyEmailType {
+  username?: string;
+  email: string;
+  language?: string;
+}
+
+export const sendEmailVerification = async ({ email, language, username }: VerifyEmailType) => {
+  const token = randomBytes(32).toString("hex");
+  const translation = await getTranslation(language ?? "en", "common");
+  const flags = await getFeatureFlagMap(prisma);
+
+  if (!flags["email-verification"]) {
+    log.warn("Email verification is disabled - Skipping");
+    return { ok: true, skipped: true };
+  }
```

Comment:
I'd rather use the getFeatureFlagMap on server-side instead of calling individual features.